scanner/src/main/kotlin/scanners/fossid/FossIdConfig.kt

```diff
@@ -135,9 +135,9 @@ internal data class FossIdConfig(
                 ?: throw IllegalArgumentException("No FossId User configuration found.")
             val packageNamespaceFilter = fossIdScannerOptions[NAMESPACE_FILTER_PROPERTY].orEmpty()
             val packageAuthorsFilter = fossIdScannerOptions[AUTHORS_FILTER_PROPERTY].orEmpty()
-            val addAuthenticationToUrl = fossIdScannerOptions[CREDENTIALS_IN_URL_PROPERTY].toBoolean()
-            val waitForResult = fossIdScannerOptions[WAIT_FOR_RESULT_PROPERTY]?.toBooleanStrictOrNull() ?: true
-            val deltaScans = fossIdScannerOptions[DELTA_SCAN_PROPERTY].toBoolean()
+            val addAuthenticationToUrl = fossIdScannerOptions[CREDENTIALS_IN_URL_PROPERTY]?.toBoolean() ?: false
+            val waitForResult = fossIdScannerOptions[WAIT_FOR_RESULT_PROPERTY]?.toBoolean() ?: true
```

Comment:
This is wrong. If the value is `null`, `toBoolean` returns `false`. And we want the default value to be `true`.

Suggestion: `?.lowercase()?.toBooleanStrictOrNull() ?: true`

Reply:
Ah I see the `?.toBoolean ?: true`

Reply:
> If the value is `null`, `toBoolean` returns `false`.

Note my use of `?.`. This now calls the non-null version of `String.toBoolean()`, not anymore `String?.toBoolean()`, to make the default value in case of `null` explicit after the `?:`.